### PR TITLE
🐛 Protect certain attributes in story-ads

### PR DESCRIPTION
**Pull request by calebcordry**
There are certain attributes for an amp-ad shown in a story that we do not want to be defined by user input.

### CODE REVIEW

File: extensions/amp-story/0.1/amp-story-auto-ads.js [hunk 1]
```diff
@@ -252,20 +252,45 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
...
   }
 
 
+  /**
+   * @param {string} type
+   * @return {boolean}
+   * @private
+   */
+  isTypeAllowed_(type) {
+    const allowedTypes = map({
```
Comment (lannka) [inline extensions/amp-story/0.1/amp-story-auto-ads.js]: make it a static const

File: extensions/amp-story/0.1/amp-story-auto-ads.js [hunk 2]
```diff
@@ -252,20 +252,45 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
...
+      'custom': true,
+    });
+
+    return allowedTypes[type] || false;
```
Comment (lannka) [inline extensions/amp-story/0.1/amp-story-auto-ads.js]: return !!allowedTypes[type]

File: extensions/amp-story/0.1/amp-story-auto-ads.js [hunk 3]
```diff
@@ -252,20 +252,45 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
...
-    const defaultAttrs = {
+    const requiredAttrs = {
       'id': 'i-amphtml-story-ad',
       'layout': 'fill',
     };
 
     const configAttrs = this.config_['ad-attributes'];
+
+    ['height', 'width', 'layout'].forEach(attr => {
+      if (configAttrs[attr]) {
```
Comment (lannka) [inline extensions/amp-story/0.1/amp-story-auto-ads.js]: configAttrs[attr] !== undefined

File: extensions/amp-story/0.1/amp-story-auto-ads.js [hunk 4]
```diff
@@ -252,20 +252,45 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
    * @private
    */
   createAdElement_() {
-    const defaultAttrs = {
+    const requiredAttrs = {
       'id': 'i-amphtml-story-ad',
```
Comment (lannka) [inline extensions/amp-story/0.1/amp-story-auto-ads.js]: fixed ID?

File: extensions/amp-story/0.1/amp-story-auto-ads.js (hunk 3 above)
Comment (calebcordry) [reply extensions/amp-story/0.1/amp-story-auto-ads.js]: go

File: extensions/amp-story/0.1/amp-story-auto-ads.js (hunk 4 above)
Comment (calebcordry) [reply extensions/amp-story/0.1/amp-story-auto-ads.js]: oops should be class

File: extensions/amp-story/0.1/amp-story-auto-ads.js (hunk 1 above)
Comment (calebcordry) [reply extensions/amp-story/0.1/amp-story-auto-ads.js]: done.

File: extensions/amp-story/0.1/amp-story-auto-ads.js (hunk 2 above)
Comment (calebcordry) [reply extensions/amp-story/0.1/amp-story-auto-ads.js]: done

**Pull request merged**